### DISCUSSION TITLE
Pgxpool, how to set Max Idle Connections setMaxIdleConns()?

### DISCUSSION THREAD

**Discussion by arkmech** [jackc/pgx#1640] [Q&A] [answered]
Stdlib, you can set the Max Idle Connections with `setMaxIdleConns()`

There doesn't seem to be a method for setting max idle connections with `pgxpool`.

I am setting other pgxpool Config, like Max Connections like this:
`db.Config().MaxConns = int32(cfg.db.maxOpenConns)`

What I expect (doesn't exist):
`db.Config().MaxIdleConns = int32(cfg.db.maxIdleConns)`
Based on docs, unless I'm blind.

**Comment by jackc** [comment]
You are correct. That doesn't exist. pgxpool's method for limiting idle connections is `MaxConnIdleTime`.

**Reply by arkmech** [reply]
Ok, do you plan on adding MaxIdleConns in the future? MaxIdleConns and MaxConnIdleTime are different parameters (not synonymous).

**Reply by jackc** [reply, marked as answer]
> Ok, do you plan on adding MaxIdleConns in the future?

I have no plans personally to add it. But I'm not particularly against the feature if someone else wants to build it.

> MaxIdleConns and MaxConnIdleTime are different parameters (not synonymous).

True, but the goal of avoiding too many idle connections existing for too long can still be achieved.
